lametro/templates/partials/past_event_day.html

```diff
@@ -63,7 +63,7 @@ <h5 class="text-default">
         {% endif %}
     </div>
     <div class="col-xs-1">
-        {% with minutes=event.board_event_minutes %}
+        {% with minutes=event.minutes.0.links.all.0.url %}
```

Comment:
This is a long chain of index and attribute access. Could it be moved up into a revised `board_event_minutes` attribute of the Event model? Also, is it safe to assume that the first link to the first minutes file will always be what we want here?

Reply:
doing this way takes advantage of the prefetching. Doing it as view or model attributes leads to a lot of queries and really slows this page down a lot. Maybe that's fine if we are caching.

Reply:
lmk.

Reply:
[this article](https://blog.mounirmesselmeni.de/2017/02/06/accessing-prefetched-objects-inside-your-models-methods/) suggests a way of taking advantage of prefetching while having method encapsulation. It's not pretty, but might be an option if we care about non-cached performance.

Reply:
Ah yes, that article is an old frenemy... What if you moved the document prefetching into a manager method akin to `with_media` and used that manager method in a revised `board_event_minutes` attribute? (Do let me know if this is starting to feel like beating a dead  horse – appreciate you taking the time to talk through this with me!)

Reply:
Oh, no, that won't work, will it – you'd still call the prefetch it for every event. Ok, I can tolerate the template code. Again, thanks for humoring me.